services/common/src/main/java/org/collectionspace/services/common/storage/JDBCTools.java

```diff
@@ -601,40 +601,27 @@ public static void grantPrivilegeToDatabaseUser(String dataSourceName, String re
      */
     public static void createNewDatabaseUser(String dataSourceName, String repositoryName,
              String cspaceInstanceId, DatabaseProductType dbType, String username, String userPW) throws Exception {
-        Statement stmt = null;
-        Connection conn = null;
         if (dbType != DatabaseProductType.POSTGRESQL) {
             throw new UnsupportedOperationException("createNewDatabaseUser only supports PostgreSQL");
         }
-        try {
+
+        String sql = null;
+        try (Connection conn = getConnection(dataSourceName, repositoryName, cspaceInstanceId);
+             Statement stmt = conn.createStatement()) {
             if (hasDatabaseUser(dataSourceName, repositoryName, cspaceInstanceId, dbType, username)) {
-                if (logger.isDebugEnabled()) {
-                    logger.debug("User: " + username + " already exists.");
-                }
+                logger.debug("User: {} already exists.", username);
             } else {
-                conn = getConnection(dataSourceName, repositoryName, cspaceInstanceId);
-                stmt = conn.createStatement();
-                String sql = "CREATE ROLE " + username + " WITH PASSWORD '" + userPW + "' LOGIN";
+                sql = "CREATE ROLE " + username + " WITH PASSWORD '" + userPW + "' LOGIN";
                 stmt.executeUpdate(sql);
-                if (logger.isDebugEnabled()) {
-                    logger.debug("Created User: " + username);
-                }
+                logger.debug("Created User: {}", username);
             }
         } catch (Exception e) {
-            logger.error("createNewDatabaseUser failed on exception: " + e.getLocalizedMessage());
-            throw e;
-        } finally {
-            try {
-                if (stmt != null) {
-                    stmt.close();
-                }
-                if (conn != null) {
-                    conn.close();
-                }
-            } catch (SQLException sqle) {
-                // nothing we can do here except log
-                logger.warn("SQL Exception when closing statement/connection: " + sqle.getLocalizedMessage());
+            logger.error("createNewDatabaseUser failed with exception:", e);
+            if (sql != null) {
+                logger.error("The following SQL statement failed using credentials from datasource named '{}': '{}'",
+                             dataSourceName, sql);
```

Comment:
I'm not comfortable logging a user-submitted password. I think I'd rather not know the SQL statement than possibly accidentally see someone's intended password. Even if the user failed to be created now, the password might be reused when it succeeds.

Reply:
When looking at this initially I'm pretty sure I mixed up how this query is made w/ a PreparedStatement. Given that this is an error level too it's almost guaranteed to show up in the logs. I'll remove it from the PR.

Honestly I'd like to change how the query is built too, I think just concatenating the Strings makes it a candidate for injection.

Reply:
That is an excellent point. It would be great to change to a PreparedStatement.

Reply:
@ray-lee While making changes I realized that this is a DDL statement and can't be used with a PreparedStatement. It's only called when creating the nuxeo database, and it seems to be getting the user/password from the DataSource created by the webapp context, so it's not _really_ user input. 

Likely something to think more about though, especially in terms of how we want to handle db initialization in the future.

Reply:
Ah, ok. Let's still not log the password, in case someone who has access to the logs shouldn't have access to the datasource configuration.